### ISSUE TITLE
no encoding decl. + non-ASCII docstring → SphinxError: 'ascii' codec can't decode byte …

### ISSUE THREAD

**Issue by shimizukawa** [sphinx-doc/sphinx#1185]
If a Python module has no encoding declartion, but it contains a non-ASCII docstring, the autodoc extension causes a SphinxError:

```
#!text
Running Sphinx v1.2b1
loading pickled environment... not yet created
No builder selected, using default: html
building [html]: targets for 1 source files that are out of date
updating environment: 1 added, 0 changed, 0 removed
reading sources... [100&#37;] index                                                                                                                                
Traceback (most recent call last):
  File "/usr/lib/pymodules/python2.7/sphinx/cmdline.py", line 247, in main
    app.build(force_all, filenames)
  File "/usr/lib/pymodules/python2.7/sphinx/application.py", line 211, in build
    self.builder.build_update()
  File "/usr/lib/pymodules/python2.7/sphinx/builders/__init__.py", line 211, in build_update
    'out of date' &#37; len(to_build))
  File "/usr/lib/pymodules/python2.7/sphinx/builders/__init__.py", line 231, in build
    purple, length):
  File "/usr/lib/pymodules/python2.7/sphinx/builders/__init__.py", line 131, in status_iterator
    for item in iterable:
  File "/usr/lib/pymodules/python2.7/sphinx/environment.py", line 458, in update_generator
    self.read_doc(docname, app=app)
  File "/usr/lib/pymodules/python2.7/sphinx/environment.py", line 609, in read_doc
    raise SphinxError(str(err))
SphinxError: 'ascii' codec can't decode byte 0xf1 in position 0: ordinal not in range(128)

Sphinx error:
'ascii' codec can't decode byte 0xf1 in position 0: ordinal not in range(128)
```

I stumbled upon this bug when building [eventlet](http://eventlet.net/): one of the modules (with no encoding declaration) copied `__doc__` from a 3rd party module (which happened to have UTF-8 docstrings).

A minimal test-case is attached.

---
- Bitbucket: https://bitbucket.org/birkenfeld/sphinx/issue/1185
- Originally reported by: Jakub Wilk
- Originally created at: 2013-06-06T00:03:35.903

**Comment by shimizukawa** [comment]
_From Sergey Shepelev on 2013-08-07 15:17:39+00:00_

[Jakub Wilk](https://bitbucket.org/jwilk) zeromq guys added coding directive to that file here https://github.com/zeromq/pyzmq/commit/55f2bec0815f31fc617b94b0de4c7eae15c7736b

But it still does not fix the problem.

**Comment by shimizukawa** [comment]
_From Georg Brandl on 2013-09-16 08:39:10+00:00_

Python itself will also raise an exception when you try to import this file, so I don't see why Sphinx should not complain. But the error can be improved.

**Comment by shimizukawa** [comment]
_From Georg Brandl on 2013-09-16 08:45:14+00:00_

Closes #1185: Don't crash when a Python module has a wrong or no encoding declared,
and non-ASCII characters are included.

→ <<cset ee2c8d01c2e93f795a9e757e55fb90c950071f38>>
